pandas/tests/dtypes/test_dtypes.py

```diff
@@ -414,16 +414,17 @@ def test_construction_from_string(self, dtype):
         assert is_dtype_equal(dtype, result)
         result = PeriodDtype.construct_from_string("period[D]")
         assert is_dtype_equal(dtype, result)
-        with pytest.raises(TypeError):
+        msg = "Cannot construct a 'PeriodDtype' from "
```

Comment:
This error message is raised by pandas itself. In these cases, we would want to check the complete message.

Reply:
The reason I left it out is because it's just repeating the input that we're giving it. If that's the case, then could I create a list of all of the strings and then for-loop through them? I don't know how much I should be changing the tests to add these messages so I was hesitant to go too far.

Reply:
if you can split and parametrize this tests could then do something like .. `msg = f"Cannot construct a 'PeriodDtype' from '{string}'"`

for guidance checkout test_construction_from_string tests for IntervalDtype (test_construction_from_string_errors) and DatetimeTZDtype (test_construct_from_string_invalid_raises) that have already been split and parametrized.

Reply:
Understood, I added them using what you gave as a template. I also fixed the other two tests, everything passes.